test/test_system_data.jl

```diff
@@ -788,6 +788,56 @@ end
     end
 end
 
+@testset "Test get_component_supplemental_attribute_pairs" begin
+    data = IS.SystemData()
+    geo_supplemental_attribute = IS.GeographicInfo()
+    bus1 = Bus("bus1", true)
+    bus2 = Bus("bus2", true)
+    gen1 = ThermalGenerator("gen1", bus1, true)
+    gen2 = PVGenerator("gen2", bus2, false)
+    for component in (bus1, bus2, gen1, gen2)
+        IS.add_component!(data, component)
+        IS.add_supplemental_attribute!(data, component, geo_supplemental_attribute)
+    end
+    ca_pairs = IS.get_component_supplemental_attribute_pairs(
+        AbstractGenerator,
+        IS.SupplementalAttribute,
+        data,
+    )
+    @test length(ca_pairs) == 2
+    sort!(ca_pairs; by = x -> IS.get_name(x.component))
+    component_names = [IS.get_name(x.component) for x in ca_pairs]
+    @test component_names == ["gen1", "gen2"]
+
+    ca_pairs = IS.get_component_supplemental_attribute_pairs(
+        AbstractPowerSystemComponent,
+        IS.SupplementalAttribute,
+        data;
+        only_available_components = true,
+    )
+    @test length(ca_pairs) == 3
+    sort!(ca_pairs; by = x -> IS.get_name(x.component))
+    component_names = [IS.get_name(x.component) for x in ca_pairs]
+    @test component_names == ["bus1", "bus2", "gen1"]
+
+    ca_pairs = IS.get_component_supplemental_attribute_pairs(
+        PVGenerator,
+        IS.SupplementalAttribute,
+        data;
+        only_available_components = true,
+    )
+    @test length(ca_pairs) == 0
+
+    ca_pairs = IS.get_component_supplemental_attribute_pairs(
+        ThermalGenerator,
+        IS.SupplementalAttribute,
```

Comment:
nitpick: test with concrete supplemental attribute type

Reply:
Oops, I thought I had one of those. Thanks.